src/Components/Components/src/ErrorBoundaryBase.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// A base class for error boundary components.
+    /// </summary>
+    public abstract class ErrorBoundaryBase : ComponentBase, IErrorBoundary
+    {
+        private int _errorCount;
+
+        /// <summary>
+        /// The content to be displayed when there is no error.
+        /// </summary>
+        [Parameter] public RenderFragment? ChildContent { get; set; }
+
+        /// <summary>
+        /// The content to be displayed when there is an error.
+        /// </summary>
+        [Parameter] public RenderFragment<Exception>? ErrorContent { get; set; }
+
+        /// <summary>
+        /// The maximum number of errors that can be handled. If more errors are received,
+        /// they will be treated as fatal. Calling <see cref="Recover"/> resets the count.
+        /// </summary>
+        [Parameter] public int MaximumErrorCount { get; set; } = 100;
```

Comment:
Do we need this to be public if we're only using it to determine whether logic in `ErrorContent` is causing an infinite stream of exceptions? It's not clear to me why we would want users to be able to customize it if we can set a sensible default.

Reply:
It's certainly open for debate. A developer might, for example, have a list with 101 items and want to be able to tolerate the case where every item fails independently, so they might have a reason to increase this number.

The tradeoff with bigger values is that there's more processing and more log messages generated before we eventually stop the process. I'm not sure there would be a single value that would be ideal for all cases.